src/command_modules/azure-cli-consumption/azure/cli/command_modules/consumption/custom.py

```diff
@@ -32,41 +47,68 @@ def cli_consumption_list_usage(cmd, client, billing_period_name=None, top=None,
         filter_to = "properties/usageEnd le \'{}\'".format(end_date.strftime("%Y-%m-%dT%H:%M:%SZ"))
         filter_expression = "{} and {}".format(filter_from, filter_to)
     if top:
-        pages = client.list(scope, expand=expand, filter=filter_expression, top=top)
+        pages = client.list_by_billing_period(billing_period_name, expand=expand, filter=filter_expression, top=top)
         return list(pages.advance_page())
-    return list(client.list(scope, expand=expand, filter=filter_expression))
+    return list(client.list_by_billing_period(billing_period_name, expand=expand, filter=filter_expression))
 
 
-def cli_consumption_list_reservations_summaries(client, grain, reservationorderid, reservationid=None, start_date=None, end_date=None):
-    """List all the reservation summaries """
-    if reservationid:
-        scope = "providers/Microsoft.Capacity/reservationorders/{}/reservations/{}".format(reservationorderid, reservationid)
-    else:
-        scope = "providers/Microsoft.Capacity/reservationorders/{}".format(reservationorderid)
+def cli_consumption_list_reservations_summaries(client, grain, reservation_order_id, start_date=None, end_date=None):
+    filter_from = None
+    filter_to = None
+    filter_expression = None
+    if start_date and end_date:
+        filter_from = "properties/UsageDate ge {}".format(start_date.strftime("%Y-%m-%dT%H:%M:%SZ"))
+        filter_to = "properties/UsageDate le {}".format(end_date.strftime("%Y-%m-%dT%H:%M:%SZ"))
+        filter_expression = "{} and {}".format(filter_from, filter_to)
+        return list(client.list_by_reservation_order(reservation_order_id, grain=grain, filter=filter_expression))
+
+    return list(client.list_by_reservation_order(reservation_order_id, grain=grain))
 
+
+def cli_consumption_list_reservations_summaries_reservation_id(client, grain, reservation_order_id, reservation_id=None, start_date=None, end_date=None):
     filter_from = None
     filter_to = None
     filter_expression = None
     if start_date and end_date:
         filter_from = "properties/UsageDate ge {}".format(start_date.strftime("%Y-%m-%dT%H:%M:%SZ"))
         filter_to = "properties/UsageDate le {}".format(end_date.strftime("%Y-%m-%dT%H:%M:%SZ"))
         filter_expression = "{} and {}".format(filter_from, filter_to)
-        return list(client.list(scope, grain=grain, filter=filter_expression))
+        return list(client.list_by_reservation_order_and_reservation(reservation_order_id, reservation_id, grain=grain, filter=filter_expression))
 
-    return list(client.list(scope, grain=grain))
+    return list(client.list_by_reservation_order_and_reservation(reservation_order_id, reservation_id, grain=grain))
```

Comment:
Same here, if "reservation_id" is None, does this essentially become the same as `client.list_by_reservation_order()`?